pkg/providers/vsphere/controlplane.go

```diff
@@ -52,6 +58,77 @@ func (b *ControlPlaneBuilder) BuildFromParsed(lookup yamlutil.ObjectLookup) erro
 	return nil
 }
 
+// ControlPlaneSpec builds a vsphere ControlPlane definition based on an eks-a cluster spec.
+func ControlPlaneSpec(ctx context.Context, logger logr.Logger, client kubernetes.Client, spec *cluster.Spec) (*ControlPlane, error) {
+	templateBuilder := NewVsphereTemplateBuilder(time.Now, true)
```

Comment:
I set the second parameter to `true` since this will be called from the controller, although we might be able to remove this soon since this new controller upgrades in order (control plane, etcd upgrade before worker nodes), right? @g-gaston

Reply:
It will follow that order indeed. However, do we need any check in between?
Also, I would like to understand why we made that cgroup part conditional in CLI. Even if it wasn't needed, did it cause any issues? Ideally the spec generated by the CLI and the controller should be the same.

Reply:
Ok so this conditional case was added pre-GA, specifically for 1.20 --> 1.21 upgrades: https://github.com/aws/eks-anywhere/pull/16. Soon after this, we implemented in-order upgrades by separating the cp & md capi specs & applying the cp spec first.
So it looks like we can remove it now since we upgrade in order for both CLI and the controller.

Reply:
Sweet
Do you know the implications of always setting the cgroup? for both CLI and controller

Reply:
I'm not completely sure but from my current understanding, it seems that as long as we have in order upgrades (which is true for both the CLI and new controller), we wouldn't have to set the cgroup at all because it will auto default to the desired cgroup driver, which is systemd for k8s 1.21+.
https://github.com/aws/eks-anywhere/blob/main/pkg/providers/vsphere/template.go#L82-L84

I can add details in the ticket to verify that with both CLI and controller.

Reply:
Chat offline, @mitalipaygude will address this separately